### ISSUE TITLE
map deserialization should enforce unique keys

### ISSUE THREAD

**Issue by ChrisBFX** [USCiLab/cereal#267]
I think it would be a good idea if cereal would throw an exception if during deserialization of a std::map/std::unordered_map a key is used several times.

**Comment by autosquid** [comment]
but, how could that be?

**Comment by erichkeane** [comment]
Messages aren't guaranteed to come from a reliable source.  This is definitely a situation that could happen.

**Comment by M2tM** [comment]
Yeah, if the json file is mucked with or generated outside of cereal for some reason.

**Comment by ChrisBFX** [comment]
Exactly. We are using cereal for creating and parsing json for rest-apis and we had the case that the other side used a field like a multimap. An exception would be nice to catch those misbehavior's earlier (instead of months later).
